### ISSUE TITLE
React Native fails on Android - "Invalid response for blob:"

### ISSUE THREAD

**Issue by psegalen** [apollographql/react-apollo#1961]
**Intended outcome:**

I developed an app using `react-apollo` in React Native 0.54.4 and everything works fine on iOS, no problem with the lib, the connection with my GraphQL server is OK.
Android version using the same code should be OK as well.

**Actual outcome:**

When I test the app on Android, any usage of a GraphQL query or mutation fails with a "Invalid response for blob:" error.
It seems that it's due to a bug in RN Android native implementation of XHR: when debugging I can see that the response sent back from native to the JS is empty, which isn't considered as a valid response for blob ones.
I know the issue doesn't seem related to this lib but if anyone could give me a workaround, I'd be grateful!

**How to reproduce the issue:**

Create a React Native app, connect it to a GraphQL server, test it on Android.

**Version**

* apollo-client@2.2.8
* react-apollo@2.1.3
* react-native@0.54.4

**Comment by psegalen** [comment]
@kingbin yes that's the tool I used

**Comment by kingbin** [comment]
Thanks for the response. I'm seeing this issue in the Android version of my app as well. I'm not seeing this error in the iOS version. With a little digging I'm finding the error being thrown at a blob case statement for request in react-native/Libraries/Network/XMLHttpRequest.js.
You haven't seen this issue since updating your ssl grade?

**Comment by tux-tn** [comment]
The issue is not directly related to SSLlabs grade. Android and iOS SslEngines have a different list of supported  Cipher suites. 
- [ iOS SSL Cipher Suite Values](https://developer.apple.com/documentation/security/1550981-ssl_cipher_suite_values?language=objc)
- [Android SSL Cipher Suite Values](https://developer.android.com/reference/javax/net/ssl/SSLEngine)

But it's still a good thing to implement secure ciphers, you can use this tool https://mozilla.github.io/server-side-tls/ssl-config-generator/ to generate the appropriate config and [SSLlabs]( https://www.ssllabs.com ) to test it

**Comment by kamelj** [comment]
> Thanks for the response. I'm seeing this issue in the Android version of my app as well. I'm not seeing this error in the iOS version. With a little digging I'm finding the error being thrown at a blob case statement for request in react-native/Libraries/Network/XMLHttpRequest.js.
> You haven't seen this issue since updating your ssl grade?

I have the same issue now,
Did you fix it by upgrading the ssl certificate ?

**Comment by kamelj** [comment]
> Fixed my problem, for the record: it was a SSL problem on my server, certificate was grade B an it seems that Android just mutes responses of non-"A grade" servers...

Thanks, But how you upgrade the grade of ssl certificate ?

**Comment by kamelj** [comment]
yes that is right my issue solved after upgrade my api server ssl to grade A 👍
